### PR TITLE
[SolidMechanics] Apply new factory registration mechanism

### PR DESCRIPTION
**Pull request by fredroy**
See
- #4429

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/SolidMechanics/FEM/src/sofa/component/solidmechanics/fem/init.cpp [hunk 1]
```diff
@@ -48,6 +52,12 @@ const char* getModuleVersion()
     return MODULE_VERSION;
 }
 
+void registerObjects(sofa::core::ObjectFactory* factory)
+{
+    factory->registerObjectsFromPlugin("Sofa.Component.SolidMechanics.FEM.Elastic");
+    factory->registerObjectsFromPlugin("Sofa.Component.SolidMechanics.FEM.HyperElastic");
```
Comment (bakpaul) [inline Sofa/Component/SolidMechanics/FEM/src/sofa/component/solidmechanics/fem/init.cpp:59]: Missing non uniform ?

Comment (fredroy) [reply Sofa/Component/SolidMechanics/FEM/src/sofa/component/solidmechanics/fem/init.cpp:59]: indeed

**Review by epernod — CHANGES_REQUESTED** [review]
problem on fedora:
@fredroy 
fedora_clang_options — Build OK. FIXME: 8 units, 0 scene, 88 regressions

**Comment by fredroy** [comment]
> problem on fedora: @fredroy fedora_clang_options — Build OK. FIXME: 8 units, 0 scene, 88 regressions

done

**Review by epernod — APPROVED** [review]

**Pull request merged**